sdk/servicebus/azure-servicebus/azure/servicebus/aio/management/_management_client_async.py

```diff
@@ -436,6 +441,10 @@ async def update_queue(
                 self.fully_qualified_namespace,
             )
         )
+
+        property_keyword_arguments = {key: kwargs.pop(key) for key in set(kwargs.keys()) if key in queue.keys()}
+        override_properties_with_keyword_arguments(queue, **property_keyword_arguments)
```

Comment:
Rather than have a utility function to do this - would it be more efficient to simply add `kwargs` to the `_to_internal_entity()` functions to allow for overrides during building the generated object?
Better yet - also include the popping from the line above there too, so we don't have to do the additional iteration of kwargs.

For example:
```python

    def _to_internal_entity(self, kwargs=None):
        # type: () -> InternalTopicDescription
        kwargs = kwargs or {}
        ...
        self._internal_td.support_ordering = kwargs.pop('support_ordering', self.support_ordering)
        self._internal_td.auto_delete_on_idle = kwargs.pop('auto_delete_on_idle', self.auto_delete_on_idle)
        self._internal_td.enable_partitioning = kwargs.pop('enable_partitioning', self.enable_partitioning)
        self._internal_td.entity_availability_status = kwargs.pop('availability_status', self.availability_status)
        self._internal_td.enable_express = kwargs.pop('enable_express', self.enable_express)
        self._internal_td.user_metadata = kwargs.pop('user_metadata', self.user_metadata)
        return self._internal_td
```

Reply:
aha, very good point! thanks for pointing out -- I didn't notice this method before, will try to move the logic into this method

Reply:
code updated!